elliptic-curve/src/arithmetic.rs

```diff
@@ -84,3 +84,21 @@ pub trait PrimeCurveArithmetic:
     /// Prime order elliptic curve group.
     type CurveGroup: group::prime::PrimeCurve<Affine = <Self as CurveArithmetic>::AffinePoint>;
 }
+
+/// Normalize point(s) in projective representation by converting them to their affine ones.
+pub trait Normalize: group::Curve {
+    /// Perform a batched conversion to affine representation on a sequence of projective points
+    /// at an amortized cost that should be practically as efficient as a single conversion.
+    /// Internally, implementors should rely upon `InvertBatch`.
+    /// This variation takes a const-generic array and thus does not require `alloc`.
+    fn batch_normalize_array<const N: usize>(points: &[Self; N]) -> [Self::AffineRepr; N];
+
+    /// Perform a batched conversion to affine representation on a sequence of projective points
+    /// at an amortized cost that should be practically as efficient as a single conversion.
+    /// Internally, implementors should rely upon `InvertBatch`.
+    /// This variation takes a (possibly dynamically allocated) slice and returns `FromIterator<Self::AffinePoint>`
+    /// allowing it to work with any container.
+    /// However, this also requires to make dynamic allocations and as such requires `alloc`.
+    #[cfg(feature = "alloc")]
+    fn batch_normalize_slice<B: FromIterator<Self::AffineRepr>>(points: &[Self]) -> B;
```

Comment:
Just a thought...

```suggestion
    fn batch_normalize_vec(points: &[Self]) -> Vec<Self::AffineRepr>
    where
        Self::AffineRepr: Copy + Default
    {
        let mut ret = vec![Self::AffineRepr::default(); points.len()];
        Self::batch_normalize(points, &mut ret);
        ret
    }
```

Not quite as flexible, but probably the main use case and justifies a hard dependency on `alloc`.

Reply:
`FromIterator` justifies `alloc` anyways, and it allows users to work with e.g. `Box`. I'm ok with switching, but I don't really see the reason

Reply:
I pushed a commit that does this, but it breaks https://github.com/RustCrypto/elliptic-curves/pull/971.
The reason is that Rust forces me to have an impl for the function even when `not(feature = "alloc")`: https://github.com/RustCrypto/elliptic-curves/pull/971/files#r1391796578

Reply:
I'm confused, if you use the provided method I suggested above, you don't need to write a downstream impl at all since it's provided